internal/command/common/format_wrapper.go

```diff
@@ -0,0 +1,52 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"golang.stackrox.io/kube-linter/internal/set"
+)
+
+// This block enumerates all supported formats.
+var (
+	PlainFormat    = newFormat("plain")
+	MarkdownFormat = newFormat("markdown")
+)
+var (
+	// AllSupportedFormats contains the list of all supported formats.
+	AllSupportedFormats = set.NewStringSet()
```

Comment:
Does Go have some form of dependency resolution when calling global variable initializers? To me it looks like there is nothing ensuring that this initialization has taken place at them time when `newFormat("plain")` above calls `AllSupportedFormats.Add`.

It also seems weird that we store a set of supported formats. That should be a derived property, derived from a map mapping the IDs to something semantic like a render function. In short - why not derive this from `formatToRenderFunc`?

Reply:
> Does Go have some form of dependency resolution when calling global variable initializers? To me it looks like there is nothing ensuring that this initialization has taken place at them time when newFormat("plain") above calls AllSupportedFormats.Add.

Yup, Go handles this really well. See https://golang.org/ref/spec#Program_initialization_and_execution.

> In short - why not derive this from formatToRenderFunc?

The problem is that it's shared between the `checks` and `templates` command, which have their own `formatToRenderFunc`s. I've written unit tests to ensure the keys of that map and this are in sync.